js/ui/grid_core/ui.grid_core.editing_row_based.js

```diff
@@ -31,7 +32,7 @@ export const editingRowBasedModule = {
 
                 _isDefaultButtonVisible: function(button, options) {
                     const isRowMode = this.isRowBasedEditMode();
-                    const isEditRow = options.row && options.row.rowIndex === this._getVisibleEditRowIndex();
+                    const isEditRow = options.row && options.row.key === this.option(EDITING_EDITROWKEY_OPTION_NAME);
```

Comment:
Minor: I suggest extracting this._getVisibleEditRowKey() from _getVisibleEditRowIndex() and using it here.
It can help to migrate from _getVisibleEditRowIndex to _getVisibleEditRowIndex in future

Reply:
I think it's better to use the equalByValue method to compare keys